beekeeper-api/src/test/java/com/expediagroup/beekeeper/api/service/HouseKeepingMetadataServiceTest.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2019-2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.api.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import static com.expediagroup.beekeeper.api.util.DummyHousekeepingMetadataGenerator.generateDummyHousekeepingMetadata;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.repository.HousekeepingMetadataRepository;
+
+@ExtendWith(MockitoExtension.class)
+public class HouseKeepingMetadataServiceTest {
+
+  private BeekeeperService housekeepingMetadataService;
+
+  @Mock
+  private HousekeepingMetadataRepository housekeepingMetadataRepository;
+  @Mock
+  private Specification<HousekeepingMetadata> spec;
+  @Mock
+  private Pageable pageable;
+
+  @BeforeEach
+  public void beforeEach() {
+    housekeepingMetadataService = new BeekeeperService(housekeepingMetadataRepository);
```

Comment:
Might be better to change the `BeforeEach` with `BeforeAll`. The service will not change from test to test. See https://junit.org/junit5/docs/5.0.0/api/org/junit/jupiter/api/BeforeAll.html.

Reply:
It does make sense but `BeforeAll` is a static method and when I change to it, it complains saying `Non-static field 'housekeepingEntityServiceImpl' cannot be referenced from a static context`. (It's also done with `BeforeEach` like this in CL)